external/storm-kafka-client/src/test/java/org/apache/storm/kafka/spout/KafkaSpoutRebalanceTest.java

```diff
@@ -181,16 +181,16 @@ public void spoutMustIgnoreFailsForTuplesItIsNotAssignedAfterRebalance() throws
         TopicPartition partitionThatWillBeRevoked = new TopicPartition(topic, 1);
         TopicPartition assignedPartition = new TopicPartition(topic, 2);
 
-        when(retryServiceMock.getMessageId(Mockito.any(ConsumerRecord.class)))
-            .thenReturn(new KafkaSpoutMessageId(partitionThatWillBeRevoked, 0))
-            .thenReturn(new KafkaSpoutMessageId(assignedPartition, 0));
+        when(retryServiceMock.getMessageId(Mockito.any(ConsumerRecord.class), contextMock))
+            .thenReturn(new KafkaSpoutMessageId(partitionThatWillBeRevoked, 0, contextMock))
+            .thenReturn(new KafkaSpoutMessageId(assignedPartition, 0, contextMock));
 
         //Emit a message on each partition and revoke the first partition
         List<KafkaSpoutMessageId> emittedMessageIds = emitOneMessagePerPartitionThenRevokeOnePartition(
             spout, partitionThatWillBeRevoked, assignedPartition, rebalanceListenerCapture);
 
         //Check that only two message ids were generated
-        verify(retryServiceMock, times(2)).getMessageId(Mockito.any(ConsumerRecord.class));
+        verify(retryServiceMock, times(2)).getMessageId(Mockito.any(ConsumerRecord.class), contextMock);
```

Comment:
You need to put eq around contextmock here or you'll get an error

Reply:
OK. I think I tried that but it didn't quite work. I will address the tests after I make the changes you suggest. I will keep you posted on how it goes.